Marlin/src/module/settings.cpp

```diff
@@ -270,11 +270,13 @@ typedef struct SettingsDataStruct {
   //
   // MESH_BED_LEVELING
   //
-  float mbl_z_offset;                                   // bedlevel.z_offset
-  uint8_t mesh_num_x, mesh_num_y;                       // GRID_MAX_POINTS_X, GRID_MAX_POINTS_Y
-  uint16_t mesh_check;                                  // Hash to check against X/Y
-  float mbl_z_values[TERN(MESH_BED_LEVELING, GRID_MAX_POINTS_X, 3)]   // bedlevel.z_values
-                    [TERN(MESH_BED_LEVELING, GRID_MAX_POINTS_Y, 3)];
+  #if ENABLED(MESH_BED_LEVELING)
+    float mbl_z_offset;                                   // bedlevel.z_offset
+    uint8_t mesh_num_x, mesh_num_y;                       // GRID_MAX_POINTS_X, GRID_MAX_POINTS_Y
+    uint16_t mesh_check;                                  // Hash to check against X/Y
+    float mbl_z_values[TERN(MESH_BED_LEVELING, GRID_MAX_POINTS_X, 3)]   // bedlevel.z_values
+                      [TERN(MESH_BED_LEVELING, GRID_MAX_POINTS_Y, 3)];
+  #endif
```

Comment:
Our standard has been to store dummy data for common features so EEPROM doesn't necessarily require a reset after flashing new firmware with leveling turned on. This goes for various other features as well, although we have been phasing that out somewhat with the addition of new features. Unless we're prepared to wrap all the optional things like this, let's continue to follow the general standard.

If you'd like to help with the more comprehensive refactoring of the EEPROM that includes per-feature versioning and writes a more structured format, check out my proposal at https://github.com/MarlinFirmware/Marlin/projects?type=classic

Reply:
Leaving code that increases the program memory usage only to avoid an EEPROM reset after flashing a new firmware doesn't seem good to me, but maybe that could be a personal choice, especially for those with program memory space problems.